src/niweb/apps/noclook/helpers.py

```diff
@@ -970,3 +971,105 @@ def set_of_member(user, node, contact_id):
     if created:
         activitylog.create_relationship(user, relationship)
     return relationship, created
+
+def link_contact_role_for_organization(user, node, contact_handle_id, role):
+    """
+    :param user: Django user
+    :param node: norduniclient model
+    :param contact_handle_id: contact's handle_id
+    :return: contact, role: Only the role is created if it don't exists
+    """
+    role_type = NodeType.objects.filter(type='Role').first()
+
+    if six.PY2:
+        role  = role.encode('utf-8')
+
+    # delete previous relationship first
+    q = """
+        MATCH (r:Role)<-[l:Is]-(c:Contact)-[:Works_for]->(o:Organization)
+        WHERE o.handle_id = {organization_id} AND r.name = "{role_name}"
+        DELETE l RETURN c
+        """.format(organization_id=node.handle_id, role_name=role)
+    d = nc.query_to_dict(nc.graphdb.manager, q)
+
+    contact = NodeHandle.objects.get(handle_id=contact_handle_id)
+
+    role = NodeHandle.objects.get_or_create(
+            node_name = role,
+            node_type = role_type,
+            node_meta_type = 'Logical',
+            creator = user,
+            modifier = user,
+        )[0]
+
+    contact.get_node().add_role(role.handle_id)
+    contact.get_node().add_organization(node.handle_id)
+
+    return contact, role
+
+def create_contact_role_for_organization(user, node, contact_name, role):
+    """
+    :param user: Django user
+    :param node: norduniclient model
+    :param contact_name: full name of the contact
+    :return: contact, role: New objects if they're not present in the db
+    """
+    contact_type = NodeType.objects.filter(type='Contact').first()
+    role_type = NodeType.objects.filter(type='Role').first()
+
+    if six.PY2:
+        contact_name = contact_name.encode('utf-8')
+        role  = role.encode('utf-8')
+
+    # delete previous relationship first
+    q = """
+        MATCH (r:Role)<-[l:Is]-(c:Contact)-[:Works_for]->(o:Organization)
+        WHERE o.handle_id = {organization_id} AND r.name = "{role_name}"
+        DELETE l RETURN c
+        """.format(organization_id=node.handle_id, role_name=role)
+    d = nc.query_to_dict(nc.graphdb.manager, q)
+
+    first_name, last_name = contact_name.split(' ')
+
+    contact = NodeHandle.objects.get_or_create(
+            node_name = contact_name,
+            node_type = contact_type,
+            node_meta_type = 'Relation',
+            creator = user,
+            modifier = user,
+        )[0]
+
+    contact.get_node().add_property('first_name', first_name)
+    contact.get_node().add_property('last_name', last_name)
+
+    role = NodeHandle.objects.get_or_create(
+            node_name = role,
+            node_type = role_type,
+            node_meta_type = 'Logical',
+            creator = user,
+            modifier = user,
+        )[0]
+
+    contact.get_node().add_role(role.handle_id)
```

Comment:
The activitylog call is missing from the creation of new relationships (or are they done somewhere else that I am missing?). See https://github.com/SUNET/ni/blob/master/src/niweb/apps/noclook/helpers.py#L592 for an example.